_toc.yml

```
@@ -1,6 +1,7 @@
-- file: lectures/00_images_are_arrays
+format: jb-article
+root: lectures/00_images_are_arrays
+sections:
```

```suggestion
format: jb-book
root: lectures/00_images_are_arrays
chapters:
```

I think I'd like this repo to more closely resemble a book. See discussion at https://github.com/executablebooks/jupyter-book/issues/1355#issuecomment-857787907

Probably we'll need a different index page but 🤷 that can be a different PR

---

I'm not familiar with the distinctions, so I have no opinion! This PR is just the result of running the recommended jupyterbook migrate command, so the output is likely pretty generic.

While we're on the topic - I wanted to gauge interest in investigating a sphinx-based workflow instead. I've had some experience helping to set up related sites for NumPy ([site](https://numpy.org/numpy-tutorials/), [source](https://github.com/numpy/numpy-tutorials/)) and NetworkX ([site](https://networkx.org/nx-guides/), [source](https://github.com/networkx/nx-guides)), and have found the sphinx-based workflow pretty straightforward for setting things up. I know we've had a brief discussion on this elsewhere, but AIUI jupyter-book is a thin wrapper on top of the other lower-level executablebook projects that include sphinx extensions like `myst-nb`. Jupyter-book introduces a layer of indirection for things like the configuration and toc which I believe are designed to be more user-friendly for users who have never used sphinx, but are unnecessary if you are comfortable with a sphinx workflow.

Anyways - I'd be happy to put together a draft PR to illustrate what this could look like, with the understanding that it would only be for illustration purposes and it might not necessarily be adopted. I don't want to step on any toes however: if there is a strong preference for sticking with the current workflow I'm happy to leave off!

---

> While we're on the topic - I wanted to gauge interest in investigating a sphinx-based workflow instead.

I completely forgot that @alexdesiqueira has done this already! I'll switch my focus over to #59. Thanks @hmaarrfk for the reminder.

---

> Jupyter-book introduces a layer of indirection for things like the configuration and toc which I believe are designed to be more user-friendly for users who have never used sphinx

🙋 

Honestly I have never grokked sphinx despite "using it" in many projects. I cargo-cult the whole workflow, and I hate it. JB marked the first time that I felt comfortable using a sphinx-based tool.

So, I do actually have a strong preference for using JB *unless* straight sphinx gives a *concrete* advantage that we are lacking. (ie I don't want to hear "configurability", I want to hear "we want to harmonize the styling of the home page and the tutorials and JB makes that very very hard/impossible".)

---

I don't quite follow this argument.  If a tool straight out of the box gives you close to what you want, you would usually need justification for an additional dependency.

[Our docs](https://github.com/scikit-image/scikit-image/blob/main/doc/source/conf.py) will probably be the most convoluted Sphinx config you can find, but it is mainly boilerplate Sphinx config.

That all said, we don't even need Sphinx: we just have a bunch of Markdown documents, so Hugo would do the trick just as nicely, and is entirely straightforward to configure.

---

@stefanv "a bunch of markdown documents" — don't we have/want *executable* notebooks (or myst-nb) that need to be *executed* before they are published?

In the future, we probably want hidden exercise solutions, which are [in the works](https://github.com/executablebooks/jupyter-book/issues/455).

> Our docs will probably be the most convoluted Sphinx config you can find, but it is mainly boilerplate Sphinx config.

That is what I'm saying is terrible. I find jb's onboarding and configuration much more straightforward and [well-documented](https://jupyterbook.org/customize/config.html).

---

Yes, the markdown documents will then get executed using something like:

```
jupytext --to notebook --execute notebook.md
```

What I'm saying about Sphinx is that there really isn't configuration that needs to be done.  You just produce the executed markdown and compile the site.  I'm not against using another tool if it contributes something other than configuration for features we don't use.

---

I am happy to go with a "whomever does the work gets to decide" approach :)

---

Ok but it's "whoever". =P

---

Thanks, fixed.